### PR TITLE
feat: add support for test expressions

### PR DESCRIPTION
**Pull request by Kaitou786**

### CODE REVIEW

File: checks/runchecks.go [hunk 1]
```diff
@@ -43,7 +33,7 @@ func RunChecks(ctx *context.Context) []*pkg.CheckResult {
 				switch v := r.Check.(type) {
 				case v1.TestFunction:
 					tpl := v.GetTestFunction()
-					if tpl.Template == "" {
+					if tpl.Template == "" && tpl.Expression == "" {
```
Comment (moshloop) [inline checks/runchecks.go:36]:
```suggestion
					if tpl.IsEmpty() {
```

File: api/v1/common.go [hunk 2]
```diff
@@ -199,8 +199,9 @@ func (t Test) GetTestTemplate() Template {
 }
 
 type Template struct {
-	Template string `yaml:"template,omitempty" json:"template,omitempty"`
-	JSONPath string `yaml:"jsonPath,omitempty" json:"jsonPath,omitempty"`
+	Template   string `yaml:"template,omitempty" json:"template,omitempty"`
+	JSONPath   string `yaml:"jsonPath,omitempty" json:"jsonPath,omitempty"`
+	Expression string `yaml:"expression,omitempty" json:"expression,omitempty"`
```
Comment (moshloop) [inline api/v1/common.go:204]:
```suggestion
	Expression string `yaml:"expression,expr" json:"expr,omitempty"`
```

**Comment by moshloop** [comment]
Can we add a fixture demoing this ?

**Comment by Kaitou786** [comment]
[[CC] Test Expressions](https://trello.com/c/TtCuRkdg/84-cc-test-expressions)

File: checks/common.go [hunk 3]
```diff
@@ -191,3 +199,45 @@ func getNextRuntime(canary v1.Canary, lastRuntime time.Time) (*time.Time, error)
 	t := lastRuntime.Add(time.Duration(canary.Spec.Interval) * time.Second)
 	return &t, nil
 }
+
+func template(ctx *context.Context, template v1.Template) (string, error) {
+	if template.Template != "" {
+		tpl := gotemplate.New("")
+
+		funcs := gomplate.Funcs(nil)
+		funcs["humanizeBytes"] = mb
```
Comment (moshloop) [inline checks/common.go:208]: Can we move these functions to commons/template alongside jsonpath and markdownParser from kommons/ktemplate - we then a have a common base of functions to work with ?

File: checks/runchecks.go [hunk 4]
```diff
@@ -43,14 +33,14 @@ func RunChecks(ctx *context.Context) []*pkg.CheckResult {
...
-					if tpl.Template == "" {
+					if tpl.IsEmpty() {
 						break
 					}
 					message, err := template(ctx.New(r.Data), tpl)
 					if err != nil {
 						r.ErrorMessage(err)
 					} else if message != "true" {
-						r.Failf("")
+						r.Failf("expected test expression to be 'true' but got '%v'", message)
```
Comment (moshloop) [inline checks/runchecks.go:43]: I think for true or false results it shouldn't have any extra error message, if it is not true or false then you can have an error message, expecting either 'true' or 'false', got 'abc'

File: fixtures/minimal/http_fail.yaml [hunk 5]
```diff
@@ -7,3 +7,7 @@ spec:
   http:
     - endpoint: http://status.savanttools.com/?code=500
       responseCodes: [200]
+    - endpoint: http://status.savanttools.com/?code=200
+      responseCodes: [200]
+      test:
+        expr: 'sprint(code) == "500"'
```
Comment (moshloop) [inline fixtures/minimal/http_fail.yaml:13]: cant this just be `code == 500`?

**Pull request merged**